src/components/pages/profile/[name]/registration/Registration.tsx

```diff
@@ -230,11 +273,36 @@ const Registration = ({ nameDetails, isLoading }: Props) => {
                   callback={transactionsCallback}
                 />
               ),
-              complete: <Complete nameDetails={nameDetails} callback={onComplete} />,
+              complete: (
+                <Complete
+                  nameDetails={nameDetails}
+                  callback={onComplete}
+                  isMoonpayFlow={item.isMoonpayFlow}
+                />
+              ),
             }[step]
           ),
         }}
       </Content>
+      <StyledDialog
+        open={hasMoonpayModal}
+        variant="actionable"
+        onDismiss={() => {
+          if (moonpayTransactionStatus === 'waitingAuthorization') {
+            return
+          }
```

Comment:
are there any consequences for doing this ?

Reply:
If the modal gets closed at this stage, the user could potentially initiate another transaction before the moonpay backend has had a chance to update the state of the current transaction. This state only lasts a few seconds as the moonpay backend communicates with the payment provider